src/libs/ReportUtils.js

```diff
@@ -1426,28 +1426,39 @@ function buildOptimisticIOUReportAction(type, amount, currency, comment, partici
     };
 }
 
-function buildOptimisticReportPreview(reportID, iouReportID, payeeAccountID) {
-    return {
+function buildOptimisticReportPreview(chatReport, iouReport, existingReportPreviewAction, justPaidOutstandingIOU) {
+    const reportPreviewAction = existingReportPreviewAction || {
         reportActionID: NumberUtils.rand64(),
-        reportID,
-        created: DateUtils.getDBTime(),
+        reportID: chatReport.reportID,
         actionName: CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW,
         pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
-        accountID: payeeAccountID,
-        message: [
-            {
-                html: '',
-                text: '',
-                isEdited: false,
-                type: CONST.REPORT.MESSAGE.TYPE.COMMENT,
-            },
-        ],
         originalMessage: {
-            linkedReportID: iouReportID,
+            linkedReportID: iouReport.reportID,
         },
-        actorEmail: currentUserEmail,
-        actorAccountID: currentUserAccountID,
     };
+
+    reportPreviewAction.created = DateUtils.getDBTime();
+    reportPreviewAction.accountID = iouReport.managerID;
+
+    const reportAmount = CurrencyUtils.convertToDisplayString(getMoneyRequestTotal(iouReport), iouReport.currency);
+    const payerName = isPolicyExpenseChat(chatReport) ? getPolicyName(chatReport) : getDisplayNameForParticipant(iouReport.managerID || '', true);
```

Comment:
NAB we already check for undefined accountID in `getDisplayNameForParticipant`

```suggestion
    const payerName = isPolicyExpenseChat(chatReport) ? getPolicyName(chatReport) : getDisplayNameForParticipant(iouReport.managerID, true);
```